### PR TITLE
Update package versions of dtslint and ts-jest to address vulnerabilities.

### PR DESCRIPTION
**Pull request by hokeun**
Update package versions of dtslint and ts-jest to address security vulnerability warnings.

Vulnerability warnings:

> highlight.js@9.18.5 postinstall /Users/hokeunkim/Development/reactor-ts/node_modules/highlight.js
> node deprecated.js

-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*

  Verion 9 of Highlight.js has reached EOL.  It will no longer
  be supported or receive security updates in the future.
  Please upgrade to version 10 or encourage your indirect
  dependencies to do so.

  For more info:

  https://github.com/highlightjs/highlight.js/issues/2877
  https://github.com/highlightjs/highlight.js/blob/master/VERSION_10_UPGRADE.md

-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*

> reactor-ts@0.0.1 postinstall /Users/hokeunkim/Development/reactor-ts
> mkdir -p node_modules/@types/ulog; cp src/core/ulog.d.ts node_modules/@types/ulog/index.d.ts

npm notice created a lockfile as package-lock.json. You should commit this file.
added 811 packages from 847 contributors and audited 814 packages in 28.295s

40 packages are looking for funding
  run `npm fund` for details

found 2 vulnerabilities (1 low, 1 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details

Output of `npm audit fix`:

$ npm audit fix
removed 1 package in 4.388s

40 packages are looking for funding
  run `npm fund` for details

fixed 0 of 2 vulnerabilities in 814 scanned packages
  1 vulnerability required manual review and could not be updated
  1 package update for 1 vulnerability involved breaking changes
  (use `npm audit fix --force` to install breaking changes; or refer to `npm audit` for steps to fix these manually)
Hokeuns-MBP:reactor-ts hokeunkim$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
Hokeuns-MBP:reactor-ts hokeunkim$ npm audit

                       === npm audit security report ===

# Run  npm install --save-dev ts-jest@27.0.3  to resolve 1 vulnerability
SEMVER WARNING: Recommended action is a potentially breaking change
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ yargs-parser                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ ts-jest [dev]                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ ts-jest > yargs-parser                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1500                            │
└───────────────┴──────────────────────────────────────────────────────────────┘

┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ssri                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=6.0.2 <7.0.0 || >=7.1.1 < 8.0.0 || >= 8.0.1                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ dtslint [dev]                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ dtslint > @definitelytyped/utils > npm-registry-client >     │
│               │ ssri                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/565                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 2 vulnerabilities (1 low, 1 moderate) in 814 scanned packages
  1 vulnerability requires semver-major dependency updates.
  1 vulnerability requires manual review. See the full report for details.

### CODE REVIEW

**Comment by lhstrh** [comment]
The text shown above suggest that we check in `package-lock.json`. Could you push that file as well? Thanks!

**Comment by hokeun** [comment]
Thanks, Marten for pointing that out!

I realized that package-lock.json was discarded by .gitignore. I forced the commit. :)

**Comment by hokeun** [comment]
Hmm, I'm not sure if it's related to package-lock.json, but the build check started failing with weird errors. Marten, do you think I should revert the last commit?

**Comment by lhstrh** [comment]
Do the tests run OK for you locally?

File: package.json [hunk 1]
```diff
@@ -26,10 +26,10 @@
         "@babel/preset-typescript": "^7.7.7",
         "@types/jest": "^24.0.15",
```
Comment (lhstrh) [inline package.json:27]:
```suggestion
        "@types/jest": "^26.0.24",
```

File: package.json [hunk 2]
```diff
@@ -26,10 +26,10 @@
...
         "@types/node": "^13.13.5",
-        "dtslint": "^3.5.2",
+        "dtslint": "^4.1.2",
         "parsimmon": "1.13.0",
         "jest": "^24.8.0",
```
Comment (lhstrh) [inline package.json:31]:
```suggestion
        "jest": "^27.0.6",
```

**Review by lhstrh — COMMENTED** [review]
I've suggested some changes that might fix this.

**Comment by lhstrh** [comment]
I think you need to pull these changes, do `npm install` and push a new lock file.

**Comment by lhstrh** [comment]
OK, we're one step further. Could you actually push this to a branch in the `reactor-ts` repo? Then I'll address the type errors we're seeing now.

**Comment by hokeun** [comment]
I tried pushing the new lock file but it didn't work. :( 
I can check and error messages and do more debugging later.
But I think this change is not critical, so it might not be worth spending too much time on this.
If this keeps appearing, I can just discard this pull request.

**Comment by hokeun** [comment]
Sure, will do that!

**Comment by lhstrh** [comment]
Pushing the new lock file actually did work; it revealed different errors that were masked by the lock file issue.

**Comment by hokeun** [comment]
Aha, I see. I pushed the changes to the branch in origin https://github.com/icyphy/reactor-ts/tree/minor-error-fix

**Pull request merged**